src/main/java/com/mraof/minestuck/client/gui/AreaEffectScreen.java

```diff
@@ -29,63 +35,112 @@ public class AreaEffectScreen extends Screen
+		
+		this.effectTextField = new TextFieldWidget(this.font, this.width / 2 - 65, yOffset + 79, 105, 18, new StringTextComponent("Current Effect"));
+		this.effectTextField.setValue(te.getEffect().getRegistryName().toString());
+		addButton(effectTextField);
+		
+		this.effectAmplifierTextField = new TextFieldWidget(this.font, this.width / 2 + 45, yOffset + 79, 20, 18, new StringTextComponent("Current Effect Amplifier"));
+		this.effectAmplifierTextField.setValue(String.valueOf(te.getEffectAmplifier()));
+		addButton(effectAmplifierTextField);
+		
+		addButton(allMobsButton = new ExtendedButton(this.width / 2 - 65, yOffset + 105, 85, 20, getAllMobsButtonMessage(), button -> cycleIsAllMobs()));
+		
+		addButton(new ExtendedButton(this.width / 2 + 25, yOffset + 105, 40, 20, new StringTextComponent("DONE"), button -> finish()));
+	}
+	
+	private ITextComponent getAllMobsButtonMessage()
+	{
+		return this.isAllMobs ? new StringTextComponent("UNTRIGGERABLE") : new StringTextComponent("TRIGGERABLE");
+	}
+	
+	/**
+	 * Cycles between the block affecting all mobs or just players
+	 */
+	private void cycleIsAllMobs()
+	{
+		isAllMobs = !isAllMobs;
+		allMobsButton.setMessage(getAllMobsButtonMessage());
+	}
+	
+	/**
+	 * Returns the current effect type
+	 */
+	private Effect getEffect(String stringInput)
+	{
+		return ForgeRegistries.POTIONS.getValue(ResourceLocation.tryParse(stringInput));
 	}
 	
 	@Override
 	public void render(MatrixStack matrixStack, int mouseX, int mouseY, float partialTicks)
 	{
 		this.renderBackground(matrixStack);
 		RenderSystem.color4f(1F, 1F, 1F, 1F);
-		this.minecraft.getTextureManager().bind(guiBackground);
-		int yOffset = (this.height / 2) - (guiHeight / 2);
-		this.blit(matrixStack, (this.width / 2) - (guiWidth / 2), yOffset, 0, 0, guiWidth, guiHeight);
-		font.draw(matrixStack, minPosMessage, (width / 2) - font.width(minPosMessage) / 2, yOffset + 5, 0x404040);
-		font.draw(matrixStack, maxPosMessage, (width / 2) - font.width(maxPosMessage) / 2, yOffset + 40, 0x404040);
+		this.minecraft.getTextureManager().bind(GUI_BACKGROUND);
+		int yOffset = (this.height / 2) - (GUI_HEIGHT / 2);
+		this.blit(matrixStack, (this.width / 2) - (GUI_WIDTH / 2), yOffset, 0, 0, GUI_WIDTH, GUI_HEIGHT);
+		font.draw(matrixStack, MIN_POS_MESSAGE, (width / 2) - font.width(MIN_POS_MESSAGE) / 2, yOffset + 5, 0x404040);
+		font.draw(matrixStack, MAX_POS_MESSAGE, (width / 2) - font.width(MAX_POS_MESSAGE) / 2, yOffset + 40, 0x404040);
 		super.render(matrixStack, mouseX, mouseY, partialTicks);
 	}
 	
 	private void finish()
 	{
-		MSPacketHandler.sendToServer(new AreaEffectPacket(parseMinBlockPos(), parseMaxBlockPos(), te.getBlockPos()));
+		int minX = parseInt(minPosDestinationTextFieldX);
+		int minY = parseInt(minPosDestinationTextFieldY);
+		int minZ = parseInt(minPosDestinationTextFieldZ);
+		int maxX = parseInt(maxPosDestinationTextFieldX);
+		int maxY = parseInt(maxPosDestinationTextFieldY);
+		int maxZ = parseInt(maxPosDestinationTextFieldZ);
```

Comment:
Do you think you could change the position parsing to something similar to what I made for the sendificator screen? Specifically with the goal of rejecting invalid input? (as opposed to silently changing invalid coords to 0 which happens right now) You can also consider whether or not you want to change the clickability of the finish-button.

Reply:
Probably not exactly what you were suggesting, but I added the red text coloration change incorporated into other puzzle block screens

Reply:
Yep, not exactly what I was suggesting, but it does serve the purpose of rejecting invalid input and giving some feedback to the player that something is invalid.